package-res/resources/web/pentaho/lang/Base.js

```diff
@@ -153,6 +183,15 @@ define([
 
     // Used by BaseBoot.extend, just below
     BaseBoot.prototype.extend = inst_extend;
+
+    // ---
+
+    if(!baseConstructor) {
```

Comment:
I would externalise this constructor as `baseObjectConstructor`, like `baseErrorConstructor`, instead of having a default value. There is no problem changing the signature of the function, as it is private.

In the future we'll probably need a `baseArrayConstructor`, so it seems coherent.